datafusion/core/src/physical_optimizer/pruning.rs

```diff
@@ -1320,14 +1457,56 @@ fn build_statistics_expr(
             );
         }
     };
+    let statistics_expr = wrap_case_expr(statistics_expr, expr_builder)?;
```

Comment:
If there is no null_count or row_count statistics, do we still need to rewrite it?

Reply:
Theoretically, if there is no `null_count` or `row_count` statistics, we don't need to rewrite it. Practically, we don't know about `null_count` or `row_count` statistics when the rewrite takes place, because the rewrite happens in `PruningPredicate` first and then `null_count` and `row_count` statistics come in later in `PruningStatistics`

Reply:
Hmm, then I think the rules order could be changed? It sounds not making sense to rewrite predicates based on statistics before the statistics are ready.

Reply:
My bad, I think the doc I wrote was not clear, I updated the doc in https://github.com/apache/arrow-datafusion/pull/9223/commits/a8639fb8aceb12333afe99c9fd6275077a017c6e. What do you think now?

Reply:
Thank you for updating it. Looks better now.